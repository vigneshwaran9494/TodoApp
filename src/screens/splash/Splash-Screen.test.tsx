import React from 'react';
import {cleanup, render} from '@testing-library/react-native';

import SplashScreen from './Splash-screen';

afterAll(cleanup);

describe('SplashScreen', () => {
  //Should render logo
  it('should render app logo', () => {
    const logoTestIdName = 'logoTestText';
    const {getByTestId} = render(<SplashScreen />);
    const foundLogo = getByTestId(logoTestIdName);
    expect(foundLogo).toBeTruthy();
  });

  //should show app version
  it('should show version', () => {
    const versionText = 'Version 1.0';
    const {toJSON, getByText} = render(<SplashScreen />);
    const foundVersionText = getByText(versionText);
    expect(foundVersionText.props.children).toEqual(versionText);
    expect(toJSON()).toMatchSnapshot();
  });
});
