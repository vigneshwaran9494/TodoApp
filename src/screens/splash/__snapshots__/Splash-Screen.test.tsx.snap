// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplashScreen should show version 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#4527a0",
        "fontFamily": "Futura",
        "fontSize": 24,
      }
    }
    testID="logoTestText"
  >
    Todo App
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "Futura",
        "fontSize": 10,
        "marginTop": "80%",
      }
    }
  >
    Version 1.0
  </Text>
</RCTSafeAreaView>
`;
